src/components/CardForm.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import {Component, CSSProperties} from 'react';
+import {Modal, ModalHeader, ModalBody, ModalFooter, Button} from 'jsc-react-ui';
+import {Note} from './../models/Note';
+
+const styles = {
+  modal: {
+    width: 500,
+    height: 400,
+    backgroundColor: '#fff',
+    display: 'flex',
+    flexFlow: 'column',
+  },
+  input: {
+    border: 'none',
+    fontWeight: 600,
+    fontSize: '1.1em',
+    flex: 1,
+  } as CSSProperties,
+  textarea: {
+    border: 'none',
+    resize: 'none',
+    flex: 1,
+  },
+};
+
+interface Props {
+  open: boolean;
+  onSave?: (note: Note) => void;
+  onClose?: () => void;
+}
+
+export class CardForm extends Component<Props, Note> {
+
+  props: Props = {
+    open: false,
+  };
+
+  state: Note = this.getInitialState();
+
+  render() {
+    const {open} = this.props;
```

Comment:
You can destructuring others props from `this.props` and `this.state` here.

```typescript
const {open, onClose} = this.props;
const {message, title} = this.state;
```

Then update components in return statement.